libs/service/verifier_service.cpp

```diff
@@ -49,12 +49,19 @@ verify_byte_code(
     _Outptr_result_maybenull_z_ const char** error_message,
     _Out_ uint32_t* error_message_size)
 {
+    std::ostringstream error;
     const ebpf_platform_t* platform = &g_ebpf_platform_windows_service;
     std::vector<ebpf_inst> instructions{instruction_array, instruction_array + instruction_count};
     program_info info{platform};
     std::string section;
     std::string file;
-    info.type = get_program_type_windows(*program_type);
+    try {
+        info.type = get_program_type_windows(*program_type);
```

Comment:
question (not for this PR): should we change the prototype so this needn't throw an exception?

Reply:
Agree. `get_program_type_windows` today returns EbpfProgramType. If we change the prototype to return EbpfProgramType *, then this function can simply return null.

We can today also not throw an exception and instead return the EbpfProgramType for EBPF_PROGRAM_TYPE_UNSPECIFIED which will have a context descriptor with all 0s, but I don't want verifier to proceed and try to verify the program using that all zeroed out context descriptor.